### PR TITLE
Fix bug in Pruner where maps were not being marked as transitively re…

### PR DESCRIPTION
**Pull request by nayeemzen**
…achable

### CODE REVIEW

File: wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java [hunk 1]
```diff
@@ -176,7 +176,11 @@ private void markReachable() {
 
   private void mark(ProtoType type) {
     // Map key type is always scalar. No need to mark it.
```
Comment (JakeWharton) [inline wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: Can you move this comment to be above the `type = type.valueType()` line?

Comment (nayeemzen) [inline wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: will do!

File: wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java [hunk 2]
```diff
@@ -175,8 +175,12 @@ private void markReachable() {
   }
 
   private void mark(ProtoType type) {
-    // Map key type is always scalar. No need to mark it.
-    if (type.isMap()) type = type.valueType();
+    // Mark the map type as it's non-scalar and transitively reachable.
+    if (type.isMap()) {
+      marks.mark(type);
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/Pruner.java]: I wonder if this is necessary . . .

**Comment by swankjesse** [comment]
LGTM!

**Pull request merged**